src/library/Cardano/Types.hs

```diff
@@ -0,0 +1,110 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+-- |
+-- Copyright: © 2018-2020 IOHK
+-- License: Apache-2.0
+--
+module Cardano.Types
+    (
+    -- * Coin
+      Coin (..)
+    , coinIsValid
+
+    -- * UTxO
+    , UTxO (..)
+    , utxoBalance
+    , utxoPickRandom
+
+    ) where
+
+import Prelude
+
+import Control.DeepSeq
+    ( NFData (..) )
+import Crypto.Number.Generate
+    ( generateBetween )
+import Crypto.Random.Types
+    ( MonadRandom )
+import Data.Map.Strict
+    ( Map )
+import Data.Word
+    ( Word64 )
+import Fmt
+    ( Buildable (..), blockListF' )
+import GHC.Generics
+    ( Generic )
+import Numeric.Natural
+    ( Natural )
+import Quiet
+    ( Quiet (Quiet) )
+
+import qualified Data.Map.Strict as Map
+
+{-------------------------------------------------------------------------------
+                                     Coin
+-------------------------------------------------------------------------------}
+
+-- | A non-negative integer value that represents a number of Lovelace.
+--
+-- One Ada is equal to 1,000,000 Lovelace.
+--
+newtype Coin = Coin
+    { getCoin :: Word64 }
+    deriving stock (Eq, Generic, Ord)
+    deriving Show via (Quiet Coin)
+
+instance NFData Coin
+
+instance Bounded Coin where
+    minBound = Coin 0
```

Comment:
I suspect this should be `1`. I've created an issue here: https://github.com/input-output-hk/cardano-coin-selection/issues/38